php_phongo.c

```diff
@@ -2007,32 +2020,37 @@ bool php_phongo_writeresult_get_write_errors(php_phongo_writeresult_t *writeresu
 	}
 	return false;
 } /* }}} */
+
 bool php_phongo_writeresult_get_writeconcern_error(php_phongo_writeresult_t *writeresult, bson_error_t *error) /* {{{ */
 {
 	const char *err = NULL;
 	uint32_t code = 0;
 
-	if (!bson_empty0(&writeresult->write_result.writeConcernError)) {
-		bson_iter_t iter;
-
-		if (bson_iter_init_find(&iter, &writeresult->write_result.writeConcernError, "code") && BSON_ITER_HOLDS_INT32(&iter)) {
-			code = bson_iter_int32(&iter);
-		}
-		if (bson_iter_init_find(&iter, &writeresult->write_result.writeConcernError, "errmsg") && BSON_ITER_HOLDS_UTF8(&iter)) {
-			err = bson_iter_utf8(&iter, NULL);
+	bson_iter_t iter;
+	bson_iter_t citer;
+	if (!bson_empty0 (&writeresult->write_result.writeConcernErrors) &&
+			bson_iter_init (&iter, &writeresult->write_result.writeConcernErrors) &&
+			bson_iter_next (&iter) &&
+			BSON_ITER_HOLDS_DOCUMENT (&iter) &&
+			bson_iter_recurse (&iter, &citer)) {
+		while (bson_iter_next (&citer)) {
+			if (BSON_ITER_IS_KEY (&citer, "errmsg")) {
+				err = bson_iter_utf8 (&citer, NULL);
+			} else if (BSON_ITER_IS_KEY (&citer, "code")) {
+				code = bson_iter_int32 (&citer);
+			}
```

Comment:
Yup. I believe you originally wrote this for extracting the first write error. I just adapted it for the write concern. The `while` is just there to be flexible about the error document's field ordering AFAICT.

@derickr pointed out that we don't have tests for this stuff, so I'm going to add some.